src/aktualizr_repo/repo.cc

```diff
@@ -0,0 +1,128 @@
+#include "repo.h"
+#include <ctime>
+#include <regex>
+#include "crypto/crypto.h"
+
+Repo::Repo(boost::filesystem::path path, const std::string &expires) : path_(std::move(path)) {
+  expiration_time_ = getExpirationTime(expires);
+}
+
+Json::Value Repo::signTuf(const Json::Value &json, const std::string &key) {
+  std::string b64sig = Utils::toBase64(Crypto::Sign(KeyType::kRSA2048, nullptr, key, Json::FastWriter().write(json)));
+  Json::Value signature;
+  signature["method"] = "rsassa-pss";
+  signature["sig"] = b64sig;
+
+  Json::Value signed_data;
+  signature["keyid"] = Crypto::getKeyId(key);
+
+  signed_data["signed"] = json;
+  signed_data["signatures"].append(signature);
+  return signed_data;
+}
+
+std::string Repo::getExpirationTime(const std::string &expires) {
+  if (expires.size() != 0) {
+    std::smatch match;
+    std::regex time_pattern("\\d{4}\\-\\d{2}\\-\\d{2}T\\d{2}:\\d{2}:\\d{2}Z");  // NOLINT(modernize-raw-string-literal)
+    if (!std::regex_match(expires, time_pattern)) {
+      throw std::runtime_error("Expiration time has wrong format");
```

Comment:
It would be helpful to print the expected format sot that the user can fix it. ISO 8601 UTC?